File: FinalProject/app/src/main/java/com/example/finalproject/ManaReservoir.java
package com.example.finalproject;

public class ManaReservoir extends Component {
    private final int volume;

    public ManaReservoir(String name, int volume) {
        super(name);
        this.volume = volume;
    }

    public int getVolume() {
        return volume;
    }
}

File: FinalProject/app/src/main/java/com/example/finalproject/service/classes/spell/ManaReservoir.kt
package com.example.finalproject.service.classes.spell

class ManaReservoir(val volume:Double, name: String, available: Boolean): Component(name, available) {
    constructor(manaReservoir: ManaReservoir) : this(manaReservoir.volume, manaReservoir.name, manaReservoir.available)
}